### PR TITLE
Process events with BackgroundProcessor

### PR DESCRIPTION
**Pull request by jkczyz**
Update event handling based on https://github.com/rust-bitcoin/rust-lightning/pull/920.

### CODE REVIEW

**Comment by valentinewallace** [comment]
Guess we should finalize this now so the Rust samples in the docs can be more finalized as well? It pretty much LGTM besides being updated to the very latest.

File: src/main.rs [hunk 1]
```diff
@@ -521,9 +490,8 @@ async fn start_ldk() {
 		}
 	});
 
-	// Step 15: Initialize LDK Event Handling
```
Comment (valentinewallace) [inline src/main.rs:524]: Could we rename these steps to match the Build Node guide?

**Review by valentinewallace — APPROVED** [review]
utACK

**Comment by jkczyz** [comment]
Successfully connected to a peer, opened a channel, and sent a payment.

**Pull request merged**